bridge-stp.in

```diff
@@ -25,18 +25,24 @@ net_dir=/sys/class/net
 # the kernel's STP implementation.  If empty, MSTP will be used for all bridges.
 MSTP_BRIDGES=''
 
-# Set $pid to pids from /var/run* for {program}.  $pid should be declared
-# local in the caller.
+# Set $pid to pids from /var/run* for mstpd.  $pid should be declared local in
+# the caller.
 # Returns LSB exit code for the 'status' action.
 checkpid()
 {
     pid=
-    if [ -f "$1" ] ; then
+    if [ -f "$pid_file" ]; then
         local line p
         read line < "$pid_file"
         for p in $line; do
-            if [ "$p" = "${p%[!0-9]*}" ] && [ -d "/proc/$p" ]; then
-                pid="$pid $p"
+            # if [ -n "$p" ] && [ "$p" is a number ]; then
+            if [ -n "$p" ] && [ "$p" = "${p%[!0-9]*}" ]; then
+                # Make sure a process with this PID is running, that its command
+                # line contains 'mstpd', and that it is not the grep command we
+                # are using to look for the 'mstpd' string.
+                if ps -o command -ww "$p" | grep -v grep | grep 'mstpd' >/dev/null 2>&1; then
```

Comment:
`pidof` is not a function.  It is a binary that is typically included with sysvinit-utils.  It should be installed on most systems, but it may not always be installed.

I can switch this to use `pidof` if you would prefer that.